stdlib/ctypes/__init__.pyi

```diff
@@ -181,6 +181,9 @@ class c_bool(_SimpleCData[bool]):
 if sys.platform == "win32":
     class HRESULT(_SimpleCData[int]): ...  # TODO undocumented
 
+if sys.version_info >= (3, 12):
+    c_time_t: TypeAlias = c_int32 | c_int64  # noqa: Y042
```

Comment:
Unless this is literally an instance of `types.UnionType` at runtime, wouldn't it be better to do something like

```suggestion
    c_time_t: type[c_int32 | c_int64]
```

?